packages/react/src/components/button.tsx

```diff
@@ -106,128 +104,104 @@ export function UploadButton<
 
   const fileInputRef = useRef<HTMLInputElement>(null);
   const labelRef = useRef<HTMLLabelElement>(null);
-  const [uploadProgressState, setUploadProgress] = useState(
+  const [uploadProgress, setUploadProgress] = useState(
     $props.__internal_upload_progress ?? 0,
   );
-  const [files, setFiles] = useState<File[]>([]);
```

Comment:
i think it makes sense for useUploadThing to control state? Especially if the `getInputProps` should have an onChange that sets files. Felt weird passing those in like `getInputProps({ mode, files, setFiles })`. By letting the hook own the state we have access to it in `getInputProps` for "free"

Reply:
Have to think how it's used in forms though